dvc/external_repo.py

```diff
@@ -77,6 +80,13 @@ def _external_repo(url=None, rev=None, cache_dir=None):
     return new_path
 
 
+def _git_checkout(repo_path, revision):
+    from dvc.scm import Git
+
+    git = Git(repo_path)
+    git.checkout(revision)
```

Comment:
We should probably close is explicitly or we might fail removing it later on Windows.

Reply:
What do you mean by closing it, @Suor ?

Reply:
@pared You still need to close it `git.checkout` fails. Add `try&finally`